src/transformers/optimization_tf.py

```diff
@@ -58,27 +58,40 @@ def get_config(self):
         }
 
 
-def create_optimizer(init_lr, num_train_steps, num_warmup_steps, end_lr=0.0, optimizer_type="adamw"):
+def create_optimizer(
+    init_lr,
+    num_train_steps,
+    num_warmup_steps,
+    min_lr_ratio=0.0,
+    adam_epsilon=1e-8,
+    weight_decay_rate=0.0,
+    include_in_weight_decay=None,
+):
     """Creates an optimizer with learning rate schedule."""
     # Implements linear decay of the learning rate.
     lr_schedule = tf.keras.optimizers.schedules.PolynomialDecay(
-        initial_learning_rate=init_lr, decay_steps=num_train_steps, end_learning_rate=end_lr,
+        initial_learning_rate=init_lr,
+        decay_steps=num_train_steps - num_warmup_steps,
+        end_learning_rate=init_lr * min_lr_ratio,
     )
     if num_warmup_steps:
         lr_schedule = WarmUp(
             initial_learning_rate=init_lr, decay_schedule_fn=lr_schedule, warmup_steps=num_warmup_steps,
         )
+    if weight_decay_rate > 0.0:
+        optimizer = AdamWeightDecay(
+            learning_rate=lr_schedule,
+            weight_decay_rate=weight_decay_rate,
+            beta_1=0.9,
+            beta_2=0.999,
+            epsilon=adam_epsilon,
+            exclude_from_weight_decay=["LayerNorm", "layer_norm", "bias"],
+            include_in_weight_decay=include_in_weight_decay,
+        )
+    else:
+        optimizer = tf.keras.optimizers.Adam(learning_rate=lr_schedule, epsilon=adam_epsilon)
 
-    optimizer = AdamWeightDecay(
-        learning_rate=lr_schedule,
-        weight_decay_rate=0.01,
-        beta_1=0.9,
-        beta_2=0.999,
-        epsilon=1e-6,
-        exclude_from_weight_decay=["LayerNorm", "layer_norm", "bias"],
-    )
-
-    return optimizer
+    return optimizer, lr_schedule
```

Comment:
This is also a breaking change - we should document this well so that the user knows

Reply:
Done!

Reply:
Indeed, this is not backwards compatible. @jplu, do you expect this method to be currently used by users outside of the `Trainer`? Would this breaking change impact those users?

Reply:
Honestly no, I not expecting users to use this outside the TF Trainer. Also the TF trainer has been updated to use this new return format, such as the PT one.

Reply:
Right, I thought so too!